### PR TITLE
Set ecmaVersion: 9, extend prettier config

### PR DESCRIPTION
**Pull request by oddirmeyer**
I'm putting this out for others to try and provide feedback on. The auto-format battle has gotten a little worse for me since enabling auto-save features in VSCode. While it has been nice to just let the rules automatically apply, the biggest issue I've noticed in projects which import this common config is that they aren't properly set up to prevent ESLint and Prettier collisions. That is, if I explicitly save a file 10 times in a row, it will toggle back and forth between ESLint-formatted output and Prettier-formatted output. There are also some syntax things being flagged in the IDE as an error that shouldn't be (spread operators, etc).

These changes should address the issues above and, in general, prefer the out-of-the-box Prettier config over a long list of specific rule overrides.

* Extend prettier config for ESLint to suppress ESLint-formatting, avoiding collisions on auto-save.
* Remove specific rules in favor of defaults.
* Set `ecmaVersion: 9` (2018) to support spread operator syntax.
* Update required `devDependencies` listed in README.md for projects importing this configuration.

I'm working with this in houndstooth-service with good results so far and will put out a PR soon which does a full, formatting-only pass over the entire project to baseline with these changes once approved. To adopt this and clean up other projects:

1. Follow the updated steps in the README.md file in this project to install devDependencies.
1. npm link this project
1. Ensure your IDE has been set up to automatically format on save.
1. Run `npx prettier --write '**/*.js'` in your project to baseline all source code files based on these rules

Let me know of any questions or concerns.

Note: Frontend projects may need a little more care and feeding so we'll want to discuss where and how to manage any special rules needed for some projects.

### CODE REVIEW

**Comment by dhjonesclc** [comment]
Nice! Thanks for digging into this @oddirmeyer .  I put a story in the backlog for grooming tomorrow to discuss these issues and the other ones you put up on the #ffffff board.

File: README.md [hunk 1]
```diff
@@ -2,11 +2,14 @@
 ES Lint Config
 
 To add this file to your own project, setup your project's `package.json` to include
```
Comment (dhjonesclc) [inline README.md:4]: typo: 'should _use_ the' ...?

Comment (oddirmeyer) [reply README.md:4]: Thanks! Fixed

**Comment by oddirmeyer** [comment]
> Nice! Thanks for digging into this @oddirmeyer . I put a story in the backlog for grooming tomorrow to discuss these issues and the other ones you put up on the #ffffff board.

Ok, cool. Yeah, between the noise in the last few PRs I've done and the erratic auto-formatting behavior I was battling, I just decided I didn't want to wait any longer.

**Comment by dhjonesclc** [comment]
Should we also require a `.prettierignore` file in a project so we skip formatting in dirs like `public` or ? We have that referenced in our wiki docs where it talks about adding a format command to the npm scripts.

**Comment by oddirmeyer** [comment]
> Should we also require a `.prettierignore` file in a project so we skip formatting in dirs like `public` or ? We have that referenced in our wiki docs where it talks about adding a format command to the npm scripts.

I'd be okay with that but I was thinking that sort of thing would be more case-by-case depending on the project since each project is still responsible for setting up commands (if needed) and may not have directories to exclude. If we want to treat as required for consistency though, I think that's fine.

**Comment by dhjonesclc** [comment]
> > Should we also require a `.prettierignore` file in a project so we skip formatting in dirs like `public` or ? We have that referenced in our wiki docs where it talks about adding a format command to the npm scripts.
> 
> I'd be okay with that but I was thinking that sort of thing would be more case-by-case depending on the project since each project is still responsible for setting up commands (if needed) and may not have directories to exclude. If we want to treat as required for consistency though, I think that's fine.

I agree with what you are saying. I shouldn't have said "require", the ignore file, I was just thinking that we might want that in some projects as you suggested. 👍

**Review by dhjonesclc — APPROVED** [review]
Thanks Ryan!

**Review by zsidnam — APPROVED** [review]
Thanks for looking into this! I'm down to get this merged and start using it, then see if there are any things we need to adjust after using it for a while.

Visual Review ✅

**Pull request merged**